splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsAdmin.java

```diff
@@ -424,6 +435,124 @@ public static void DROP_TABLE_STATISTICS(String schema, String table) throws SQL
+            outputResults[0] = new EmbedResultSet40(conn, resultsToWrap, false, null, true);
+        } catch (StandardException se) {
+            throw PublicAPI.wrapStandardException(se);
+        }
+    }
+
+    public static void FAKE_COLUMN_STATISTICS(String schema,
+                                              String table,
+                                              String column,
+                                              double nullCountRatio,
+                                              long rpv,
+                                              ResultSet[] outputResults) throws SQLException {
+        EmbedConnection conn = (EmbedConnection) SpliceAdmin.getDefaultConn();
+        try {
+            schema = EngineUtils.validateSchema(schema);
+            table = EngineUtils.validateTable(table);
+            column = EngineUtils.validateColumnName(column);
+            TableDescriptor td = verifyTableExists(conn, schema, table);
+            //verify that that column exists
+            int columnId = -1;
+            ColumnDescriptor columnDescriptor = null;
+            ColumnDescriptorList columnDescriptorList = td.getColumnDescriptorList();
+            for (ColumnDescriptor descriptor : columnDescriptorList) {
+                if (descriptor.getColumnName().equalsIgnoreCase(column)) {
+                    columnId = descriptor.getPosition();
+                    columnDescriptor = descriptor;
+                    break;
+                }
+            }
+            if (columnId == -1)
+                throw ErrorState.LANG_COLUMN_NOT_FOUND_IN_TABLE.newException(column, schema + "." + table);
+            List<TableDescriptor> tds = Collections.singletonList(td);
+            authorize(tds);
+            DataDictionary dd = conn.getLanguageConnection().getDataDictionary();
+            dd.startWriting(conn.getLanguageConnection());
+            TransactionController tc = conn.getLanguageConnection().getTransactionExecute();
+            tc.elevate("statistics");
+            // get the row count from table stats
+            long totalCount = getRowCountFromTableStats(td.getHeapConglomerateId(), dd, tc);
+
+            if (totalCount < 0)
+                throw ErrorState.LANG_INVALID_FAKE_STATS.newException("column", "table stats do not exist, please add table stats first");
+
+            if (nullCountRatio < 0 || nullCountRatio > 1)
+                throw ErrorState.LANG_INVALID_FAKE_STATS.newException("column", "null count ratio should be in the range of [0,1]");
+
+            long nullCount = (long)(nullCountRatio * totalCount);
+
+            if (rpv > totalCount - nullCount)
+                throw ErrorState.LANG_INVALID_FAKE_STATS.newException("column", "rows per value should be larger than the total number of not-null count : " + (totalCount - nullCount));
+
+            long cardinality = (long)(((double)(totalCount- nullCount))/rpv);
```

Comment:
Might want to check for zero or negative rpv here and throw a descriptive error message, like in the other cases above.

Reply:
Done.